library/coretests/benches/fmt.rs

```diff
@@ -162,3 +162,66 @@ fn write_u8_min(bh: &mut Bencher) {
         black_box(format!("{}", black_box(u8::MIN)));
     });
 }
+
+#[bench]
+fn write_12ints_bin(bh: &mut Bencher) {
+    bh.iter(|| {
+        black_box(format!("{:b}", black_box(0_u8)));
+        black_box(format!("{:b}", black_box(100_i8)));
+        black_box(format!("{:b}", black_box(-100_i8)));
+
+        black_box(format!("{:b}", black_box(0_u32)));
+        black_box(format!("{:b}", black_box(1000_i32)));
+        black_box(format!("{:b}", black_box(-1000_i32)));
+
+        black_box(format!("{:b}", black_box(0_u64)));
+        black_box(format!("{:b}", black_box(10000_i64)));
+        black_box(format!("{:b}", black_box(-10000_i64)));
+
+        black_box(format!("{:b}", black_box(0_u128)));
+        black_box(format!("{:b}", black_box(100000_i128)));
+        black_box(format!("{:b}", black_box(-100000_i128)));
+    });
+}
```

Comment:
I think it would be good to break up benchmarks by integer size because that can be pretty platform-dependent, e.g. u64 might be about the same as u32 on 64-bit platforms, but significantly slower on 32-bit.

Testing a few assorted values seems good though. To get an even balance it may also be good to check:

* `1 << n` for n `0..U::BITS`, both signed and unsigned
* `10.pow(n)` for `n = 0` up to the max that fits
* `I::MIN`, `I::MAX`, `U::MAX`

These can be collected outside of `bb.iter` so the cost of computing values isn't included in the benchmarks

Reply:
This is preexisting but it would also be better to create a `String::with_capacity(enough_for_u128)`, within `bb.iter` just `.clear()` and `write!` into it. Keeps time to alloc/free from skewing benchmark results.

Reply:
I think there is some advantage to having a bit of a wider range to attempt to reflect something for cases where e.g. formatting slows down for values above a specific count. But it doesn't need to be as extensive as I suggested.

> A sink might be better than buffering with String?

I think the formatter might not actually get called with `Sink` https://github.com/rust-lang/rust/blob/4ff3fa01cbdd468851b1b859541ee1c648cde7de/library/std/src/io/util.rs#L400-L403. Writing to a `String` is a pretty common usecase so I think that is fine.